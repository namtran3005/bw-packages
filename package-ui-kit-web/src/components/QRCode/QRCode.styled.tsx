import * as React from 'react';
import styled from 'styled-components';
import { QRCode as QRCodeSVG, QRCodeProps } from 'react-qr-svg';

type RootProps = QRCodeProps & {
  style?: object;
};

const RootComponent: React.FunctionComponent<RootProps> = ({
  ...cleanProps
}) => <QRCodeSVG {...cleanProps} />;

const Root = styled(RootComponent)`
  &&& {
    width: 100%;
  }
`;

Root.displayName = 'QRCode';

export { Root, RootProps, RootComponent };
