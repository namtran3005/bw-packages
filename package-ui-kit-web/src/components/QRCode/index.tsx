export { QRCode, QRCodeProps } from './QRCode';
