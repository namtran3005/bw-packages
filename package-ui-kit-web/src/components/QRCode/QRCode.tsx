import * as React from 'react';
import { Root, RootProps as QRCodeProps } from './QRCode.styled';

const QRCode: React.FunctionComponent<QRCodeProps> = (props) => (
  <Root {...props} />
);

export { QRCode, QRCodeProps };
