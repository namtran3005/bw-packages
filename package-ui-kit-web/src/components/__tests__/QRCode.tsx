import * as React from 'react';
import { shallow } from 'enzyme';
import { QRCode } from '../QRCode';

describe('QRCode', () => {
  it('should render menu', () => {
    const wrapper = shallow(<QRCode value="test value" />);
    expect(wrapper).toMatchSnapshot();
  });
});
