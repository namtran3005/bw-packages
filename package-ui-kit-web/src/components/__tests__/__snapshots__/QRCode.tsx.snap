// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCode should render menu 1`] = `
<QRCode
  value="test value"
/>
`;
